File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
import java.util.*;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        List<Character> set = new ArrayList<>();
        int max_value = 0;

        for (int right = 0; right < s.length(); right++) {
            char currentChar = s.charAt(right);

            if (set.contains(currentChar)) {
                // Find the index of the duplicate character
                int index = set.indexOf(currentChar);
                // Remove all elements up to the duplicate character (including it)
                for (int i = 0; i <= index; i++) {
                    if (!set.isEmpty()) { // Check if list is not empty
                        set.remove(0);
                    }
                }
            }

            set.add(currentChar);
            max_value = Math.max(max_value, set.size());
        }

        return max_value;
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.ts
function lengthOfLongestSubstring(s: string): number {

      let set = new Set();
      let res = 0;
      let l = 0;

      for(let r=0;r<s.length;r++){ 
         while(set.has(s[r])){ 
              set.delete(s[l]);
              l++;
         }
         set.add(s[r]);
         res = Math.max(r-l+1,res);
      }

      return res;




      

    
};